src/platform/plugins/shared/unified_search/public/query_string_input/query_bar_menu_panels.tsx

```diff
@@ -357,7 +350,10 @@ export function useQueryBarMenuPanels({
     storage.set(storageKey!, lang);
 
     const newQuery = { query: '', language: lang };
-    onQueryStringChange(newQuery.query);
+    onQueryChange({
```

Comment:
This is for fixing this issue: (Special thanks to @cesco-f :) )

https://github.com/user-attachments/assets/e9074edb-20ab-426b-a293-51c3770b7348